src/traces/scattermapbox/convert.js

```diff
@@ -82,6 +83,29 @@ module.exports = function convert(gd, calcTrace) {
         var circleOpts = makeCircleOpts(calcTrace);
         circle.geojson = circleOpts.geojson;
         circle.layout.visibility = 'visible';
+        if(hasCluster) {
+            circle.filter = ['!', ['has', 'point_count']];
+            opts.cluster = {
+                type: 'circle',
+                filter: ['has', 'point_count'],
+                layout: {visibility: 'visible'},
+                paint: {
+                    'circle-color': createClusterColors(trace.cluster.color, trace.cluster.step),
+                    'circle-radius': createClusterSizes(trace.cluster.size, trace.cluster.step),
+                    'circle-opacity': trace.cluster.opacity
```

Comment:
Since the `opacity` is `arrayOk`, this line should be something like:
```js
'circle-opacity': createClusterOpacities(trace.cluster.opacity, trace.cluster.step),
```